### ISSUE TITLE
Allow `PATCH` requests to update attributes present as customProperties

### ISSUE THREAD

**Issue by abcabhishek** [open-metadata/OpenMetadata#8227]
**Affected module**
Ingestion Framework/Backend

**Describe the bug**
Making `PATCH` requests on `extension` for customProperties does not allow add/update of the customProperty through the **ingestion-bot**

**To Reproduce**

- Create a customProperty from the UI
- Add a customProperty value attribute through the UI
- Try to update the customProperty (say `tableSize`) through Postman/CURL through the following request body:
```
curl --location --request PATCH 'http://localhost:8585/api/v1/tables/a8db294a-2c8f-485c-9946-0d0bbf040a34' \
--header 'Content-Type: application/json-patch+json' \
--header 'Authorization: Bearer <token>' \
--data-raw '[
    {
        "op": "add",
        "path": "/extension/someSize",
        "value":  "65GB"
    }
]'
```
This does not result in an update to the attribute

**Expected behavior**
`PATCH` requests are used when you’d want to change the partial state of the object. 
Calling the `PATCH` endpoint with the new/updated value of the attribute should allow us to update the state of the attribute

**Version:**
 - Python version: 3.8
 - OpenMetadata version: 0.12.1

**Additional context**

**Comment by sureshms** [comment]
@abcabhishek this functionality should be working. Here are the tests that pass for this functionality - https://github.com/open-metadata/OpenMetadata/blob/a731b3fdfdb7ef0eb55a468c64558ccc1f6281e6/openmetadata-service/src/test/java/org/openmetadata/service/resources/EntityResourceTest.java#L1332

Can you share the response you are getting for your patch request?

**Comment by abcabhishek** [comment]
Hi @sureshms 
Can you refer to [this](https://openmetadata.slack.com/archives/C02B6955S4S/p1666067704420249) thread on the OM Slack channel?

**Comment by abcabhishek** [comment]
@nahuelverdugo has mentioned the following there:

```
Hi 
[@abchat](https://openmetadata.slack.com/team/U038V130WKS)
, a bot can not update the extension of a table. The reason is that the extension value is overwritten during the ingestion if we don’t pass any value for the extension. The same behavior is applied to the description of the table.

[@abchat](https://openmetadata.slack.com/team/U038V130WKS)
 the 200 is ok and it is not an error, the current design does not allow the bot to update the extension
```
Ref [code](https://github.com/open-metadata/OpenMetadata/blob/main/openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityRepository.java#L1267) link that Nahuel shared

**Comment by sureshms** [comment]
@abcabhishek and @nahuelverdugo missed the **ingestion-bot** call out in the bug report. 

@abcabhishek as long as you use PATCH we should be able to allow this. In the case of PUT, REST semantics demands that you overwrite the object and for ingestion-bot, we are not allowing such overwrite. Can you confirm you want this functionality for PATCH?

**Comment by abcabhishek** [comment]
@sureshms - I can confirm that I want this functionality for PATCH

**Comment by sureshms** [comment]
If you want this functionality soon before 0.13, you can apply this patch to your code:
```
diff --git a/openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityRepository.java b/openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityRepository.java
index e3a8e1196..f07a6eee9 100644
--- a/openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityRepository.java
+++ b/openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityRepository.java
@@ -1270,8 +1270,8 @@ public abstract class EntityRepository<T extends EntityInterface> {
         return;
       }
 
-      if (updatedByBot()) {
-        // Revert changes to extension field, if being updated by a bot
+      if (updatedByBot() && operation == Operation.PUT) {
+        // Revert extension field, if being updated by a bot with a PUT request to avoid overwriting custom extension
         updated.setExtension(original.getExtension());
         return;
       }
```

**Comment by abcabhishek** [comment]
@sureshms let me try this out

**Comment by sureshms** [comment]
This issue has been fixed by the change e7a4a7a3e556d79b2f0f538d16899a5c7f99e0ec
